util-core/src/main/scala/com/twitter/io/ByteReader.scala

```diff
@@ -384,6 +398,41 @@ private class ByteReaderImpl(buf: Buf) extends ByteReader {
     ret
   }
 
+  private val unsignedLongMaxValue: BigInt = BigInt("18446744073709551615")
+
+  def readUnsignedLongBE(): BigInt = {
```

Comment:
Close. Since java doesn't have any unsigned types, we have to use the next larger type. This facilitates that conversion: https://stackoverflow.com/questions/9578639/best-way-to-convert-a-signed-integer-to-an-unsigned-long. It is not necessary in this case.

Also, BigInt provides a constructor directly from a byte array!

`BigInt(1, Array(buf.get(pos), buf.get(pos+1), buf.get(pos+2)... )`

The first argument is how to interp. So 1 for signed, and -1 for unsigned (I believe)